core/state/statedb_arbitrum.go

```diff
@@ -60,19 +66,42 @@ func StripStylusPrefix(b []byte) ([]byte, error) {
 	return b[3:], nil
 }
 
-func (s *StateDB) GetCompiledWasmCode(addr common.Address, version uint16) []byte {
-	stateObject := s.getStateObject(addr)
-	if stateObject != nil {
-		return stateObject.CompiledWasmCode(s.db, version)
+func (s *StateDB) ActivateWasm(moduleHash common.Hash, asm, module []byte) {
+	_, exists := s.activatedWasms[moduleHash]
+	if exists {
+		return
```

Comment:
Theoretically, I think there be an issue here where:

- a WASM is activated on version 1
- an owner action enables WASM version 2
- the WASM is activated again, now on version 2

If WASM versions are tied to ArbOS versions it's probably fine as those can only be upgraded in internal txs, but perhaps worth thinking about. I think this could be fixed by overwriting the activation here anyways, and having the wasmActivation journal entry store the old activation (if any), and then on revert it'd re-assign that old activation instead of deleting it altogheter if an old one existed.

Reply:
We've been designing the system with the tacit invariant that Stylus upgrades only happen during ArbOS upgrades. We should make that explicit in a comment in ArbOS, since although that pattern works, our code is simpler here and elsewhere if we just require that of the system -- which is what the DAO would be doing anyways when bumping Stylus versions.